docs/debugging-s2i.md

```diff
@@ -0,0 +1,111 @@
+
+Troubleshooting
+=================
+
+This document contains some tips and suggestions for troubleshooting the use of source-to-image ("s2i") to inject source into Docker images.
+
+Can't Find The `sti` Executable
+--------------
+
+Whether by  running `make build` or `hack/build-release.sh`, look under the `_output/local/go/bin` directory.
+
+If you have downloaded one of the official releases (see [releases](https://github.com/openshift/source-to-image/releases)), the `sti` executable is the only file in the tar archive.
+
+Security And Docker
+--------------
+
+As noted [here](https://github.com/openshift/source-to-image/#security), there are considerations to allow S2I to interact with Docker, including possibly running `sudo sti`.
+
+Image Mechanics
+--------------
+
+By default, s2i will always pull the image from the remote Docker repository rather than using the local builder image if available on the host from which you are running the `sti` command.  However, if for performance reasons you choose to leverage the `--force-pull=false` option when running the `sti` command,
+you must guard against the local builder image becoming stale and out of date.
+
+Permissions Needed During the S2I Process
+--------------
+
+The s2i process will levarage the `/tmp` directory at various stages, such as when cloning the source repository, when augmenting the Dockerfile for layered builds, during various `tar` operations that are performed, and when executing the scripts you have provided.  Invalid permissions will result in a message like this:
+
+     E0720 21:07:17.145257 04202 main.go:328] An error occurred: creating temporary directory  failed
+
+Hence, the user ID you execute the `sti` command with must have the appropriate permission to read, create, and remove files under `/tmp`.
+
+Passing In Environment Variables
+--------------
+
+If you are passing in environment variables with the -e option from the command line, and those environment variables have values with commas in them, Go command line processing will break.
+
+       $ ~/sandbox/ruby sti build --loglevel=5 -e "MAVEN_ARGS='-P some-repo,some-other-repo'" file:///home/mfojtik/sandbox/ruby openshift/ruby-20-centos7 foo
+       	 I0519 14:20:51.410089 17228 util.go:34] Using mi@mifo.sk credentials for pulling openshift/ruby-20-centos7
```

Comment:
if mi@mifo.sk is a real user's email address from the original issue, we should replace it w/ something generic.